OpenRA.Game/Traits/World/MusicPlaylist.cs

```diff
@@ -0,0 +1,135 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Linq;
+using OpenRA.GameRules;
+
+namespace OpenRA.Traits
+{
+	[Desc("Trait for music handling. Attach this to the world actor.")]
+	public class MusicPlaylistInfo : ITraitInfo
+	{
+		public readonly string StartingMusic = null;
+		public readonly bool LoopStartingMusic = false;
+
+		public object Create(ActorInitializer init) { return new MusicPlaylist(init.World, this); }
+	}
+
+	public class MusicPlaylist
+	{
+		bool installed = false;
+		bool repeat;
+
+		MusicInfo currentSong = null;
+		MusicInfo[] music;
+		MusicInfo[] random;
+
+		public MusicInfo[] GetPlaylist { get { return music; } }
+		public bool GetInstalled { get { return installed; } }
+
+		public MusicPlaylist(World world, MusicPlaylistInfo info)
+		{
+			installed = world.Map.Rules.InstalledMusic.Any();
+
+			music = world.Map.Rules.InstalledMusic.Select(a => a.Value).ToArray();
+			if (!installed)
+				return;
+
+			random = music.Shuffle(Game.CosmeticRandom).ToArray();
+
+			if (Game.Settings.Sound.MapMusic && !string.IsNullOrEmpty(info.StartingMusic) && world.Map.Rules.Music.ContainsKey(info.StartingMusic))
+			{
+				currentSong = world.Map.Rules.Music[info.StartingMusic];
+				repeat = info.LoopStartingMusic;
+			}
+			else
+			{
+				currentSong = Game.Settings.Sound.Shuffle ? random.First() : music.First();
+				repeat = Game.Settings.Sound.Repeat;
+			}
+
+			Play();
```

Comment:
No, it actually may prolong the tick, but it sounds quite nice. ( I did test it )